internal/backend/remote-state/http/client_test.go

```diff
@@ -137,7 +137,10 @@ type testHTTPHandler struct {
 func (h *testHTTPHandler) Handle(w http.ResponseWriter, r *http.Request) {
 	switch r.Method {
 	case "GET":
-		w.Write(h.Data)
+		if _, err := w.Write(h.Data); err != nil {
+			w.WriteHeader(500)
+		}
+		w.WriteHeader(200)
```

Comment:
```suggestion
		} else {
			w.WriteHeader(200)
		}
```
I know this is a test file, but in case of an error, we will be calling the `WriteHeader` twice, so we might as well put it in an else condition.